content/en/docs/security/otel-collector-configuration-best-practices.md

```diff
@@ -0,0 +1,139 @@
+---
+title: OpenTelemetry Collector Configuration Best Practices
+linkTitle: Collector config
+description:
+  Follow best practices to securely configure OpenTelemetry Collector.
+weight: 20
+---
+
+When setting up configuration for OpenTelemetry (OTel) Collector, consider the
+following practices to better secure your OTel Collector instance.
+
+<!--
+TODO: SHOULD only enable the minimum required components. What are those?
+--->
+
+## Receivers and Exporters
+
+- Receivers and Exporters can be either push or pull-based. In either case, the
+  connection established SHOULD be over a secure and authenticated channel.
+- Unused receivers and exporters SHOULD be disabled to minimize the attack
+  vector of the Collector.
+- Receivers and Exporters MAY expose buffer, queue, payload, and/or worker
+  settings via configuration parameters. If these settings are available, users
+  should proceed with caution before modifying the default configuration values.
+  Improperly setting these values may expose the OpenTelemetry Collector to
+  additional attack vectors including resource exhaustion.
+
+It is possible that a receiver MAY require the OpenTelemetry Collector run in a
+privileged mode in order to operate, which could be a security concern, but
+today this is not the case.
+
+### Safeguards against denial of service attacks
+
+Users should bind receivers' servers to addresses that limit connections to
+authorized users, so that their Collectors aren't exposed to the public internet or to wider networks than necessary.
+
+For example, if the OTLP receiver OTLP/gRPC server only has local clients, the
+`endpoint` setting SHOULD be bound to `localhost`:
+
+```yaml
+receivers:
+  otlp:
+    protocols:
+      grpc:
+        endpoint: localhost:4317
+```
+
+Generally, `localhost`-like addresses should be preferred over the `0.0.0.0`
+address. For more information, refer to the
+[CWE-1327: Binding to an Unrestricted IP Address](https://cwe.mitre.org/data/definitions/1327.html)
+definition.
```

Comment:
```suggestion
Use `localhost` addresses instead of `0.0.0.0. For more
information, refer to the
[CWE-1327: Binding to an Unrestricted IP Address](https://cwe.mitre.org/data/definitions/1327.html)
definition.
```